src/NuGet.Core/NuGet.Protocol/Resources/RawSearchResourceV3.cs

```diff
@@ -113,6 +84,68 @@ public virtual async Task<JObject> SearchPage(string searchTerm, SearchFilter fi
             throw new FatalProtocolException(Strings.Protocol_MissingSearchService);
         }
 
+        private static StringBuilder LastStringBuilder;
```

Comment:
The cached string builder seems like a pretty aggressive optimization. Does the measured performance gain warrant this complexity?

Reply:
Running the benchmarks again, without caching the string builder it's 50-65% slower, meaning it's slower than several of the string concat scenarios (when the TFM and package type lists are empty). Now I think I should look more carefully at all the other changes I made. On the other hand, 0.15ms on some of the code paths that look like they're probably not run more than once per process just isn't worth the effort.